tests/test_hetero_random_walk.cc

```diff
@@ -0,0 +1,40 @@
+
+#include <gtest/gtest.h>
+#include "hetero_graph.h"
+#include "graph.h"
+#include <torch/torch.h>
+
+using namespace gs;
+
+TEST(RandomWalk, test1)
+{
+    auto graph1 = c10::intrusive_ptr<Graph>(std::unique_ptr<Graph>(new Graph(false)));
+    auto graph2 = c10::intrusive_ptr<Graph>(std::unique_ptr<Graph>(new Graph(false)));
+    auto options = torch::TensorOptions().dtype(torch::kInt64).device(torch::kCPU);
+    int64_t indptr1[] = {0, 0, 1, 2, 3};
```

Comment:
Could you use an example where a source could have multiple edges? You may only need to check whether a seed nodes sampled one of their neighbors.

Reply:
OK, added another test